### DISCUSSION TITLE
MUX support

### DISCUSSION THREAD

**Discussion by Vintotan** [vidstack/player#815] [Q&A] [answered]
Hi,

I would like to know if there is an integration of Vidstack into the MUX ecosystem planned. Due to the repo, MUX is a supporter of this project, isn't it?

Cheers,
Tin

**Comment by mihar-22** [comment]
Hey @Vintotan!

Ye we're huge fans of Mux and would love to see improved integration over time :) What would you like to see? What's missing today?

**Reply by Vintotan** [reply]
## Integrated WebVTT support

Currently, I' using the Mux video player and a very nice feature is the [Timeline hover preview](https://www.mux.com/player) which is autogenerated when encoding video files using Mux.

## Mux Data integration

It would be very nice if there would be a Mux Data integration of the Vidstack player so that people can get analytics and metrics from the Mux Data console.

**Reply by mihar-22** [reply, marked as answer]
## Thumbnails

We already support thumbnails via the [`media-slider-thumbnail`](https://www.vidstack.io/docs/player/components/sliders/slider-thumbnail) component. Mux provides a simple URL to get the [storyboard VTT file](https://docs.mux.com/guides/video/create-timeline-hover-previews#webvtt) which looks something like this `https://image.mux.com/{PLAYBACK_ID}/storyboard.vtt`. All you need to do is paste it in the `src` of the thumbnail component like so:

```html
<media-time-slider>
  <media-slider-thumbnail src="https://image.mux.com/{PLAYBACK_ID}/storyboard.vtt" />
</media-time-slider>
```

## Mux Data

Ye we'd love to add a smooth Mux Data integration. I believe it already works on top of native video element today so you can connect it, but for a tighter and improved integration we'll need to talk to the team. 

## Mux Provider

We'll definitely look into adding a Mux provider which will simplify integration so all you'll need to do is provide a playback ID. This will most likely happen after 1.0.
